### PR TITLE
Fix table header and body column misalignment

### LINKED ISSUE #9779: Table Block: table cells in thead are misaligned when "Fixed width table cells" is toggled off
Steps to reproduce:

1. Create a new table block.
1. Edit it to use the sample HTML below which contains `thead` and `tbody` tags.
1. Switch back to "Edit Visually."
1. Toggle the "Fixed width table cells" option in the sidebar.

<strong>Result:</strong> when "Fixed width table cells" is toggled off, the `thead` and `tbody` table cells are misaligned. 

<strong>Sample HTML:</strong> 

```
<table class="wp-block-table"><thead><tr><td><strong>Column One</strong></td><td><strong>Column Two</strong></td></tr></thead><tbody><tr><td>one</td><td>two</td></tr><tr><td>three</td><td>four</td></tr></tbody></table>
```

screen shot 2018-09-10 at 8 01 26 pm
<sup>Seen at https://madefortesting.com/wp-admin/post.php?post=512&action=edit running WordPress 4.9.8 and Gutenberg 3.7 using Firefox 62.0 on macOS 10.13.6.</sup>

h/t @mcsf for the report at https://github.com/WordPress/gutenberg/pull/8767#issuecomment-417854831

**Note:** even though the alignment looks off in the editor, it looks correct on the front-end using the Twenty Seventeen theme ([screenshot](https://user-images.githubusercontent.com/1119271/45333658-9b1b8900-b534-11e8-838d-32595abcbb08.png)).

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #9779 - table header misalignment issue

Changes:
1. The issue was caused by `thead`, `tbody` and `tfoot` elements being given the `display: table` property (SO explanation - https://stackoverflow.com/a/12153220). This styling has now been removed so that the elements have their default display property.
2. `table-layout: fixed;` was previously applied on the `thead`, `tbody` and `tfoot` elements, but since they're no longer `display: table` that was no longer working. This style is now applied to the `table` element and the `table` element has been changed from `display: block` to `display: table`
3. The above changes break the `overflow-x: auto` style that was applied to the `table` element, so a new wrapping `div` has been introduced to the editor and front-end rendered output of the block. Overflow is now applied to this div.
4. Table block deprecation has been introduced since the output of the block has been changed.

## How has this been tested?
**Column alignment**
1. Paste a table with a `thead` directly into the editor (e.g. `<table><thead><tr><td><strong>Column One</strong></td><td><strong>Column Two</strong></td></tr></thead><tbody><tr><td>one</td><td>two</td></tr><tr><td>three</td><td>four</td></tr></tbody></table>`)
2. Observed in the editor that the table header columns are aligned with the body regardless of whether the 'Fixed width table cells' option is enabled.
3. Observed in the preview that everything is aligned. 

**Overflow**
1. Enter long content without spaces or line breaks in a cell to make the table wider than the editor width.
2. Observed that the user can scroll horizontally in the editor to view all the content of the table
3. Observed that the user can scroll horizontally in the preview to view the entire table. (To test this, `add_theme_support( 'wp-block-styles' );` needs to be added somewhere (e.g. in client-assets.php)).

**Fixed width table cells**
1. Toggle fixed width table cells
2. Ensured that table cells are equal spaced and that long content wraps in both the editor and preview

**Deprecation**
1. Create a table block on master branch and save post
2. Checkout this branch and load post in editor.
3. Ensured 'This block has been modified externally.' message is not present, and table displayed as expected 

## Screenshots 
**Before**
screen shot 2018-09-18 at 6 09 04 pm

**After**
screen shot 2018-09-18 at 5 57 30 pm

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/table/style.scss
Comment (talldan) [inline packages/block-library/src/table/style.scss]: @jasmussen ah, I think this should be in theme.scss instead of style.scss, is that right?

**Comment by jasmussen** [comment]
Very cool, thanks so much for working on this. Really really appreciate it.

It seems to be working:

screen shot 2018-09-19 at 08 34 44

I detected an issue with my own theme as part of this where I broke the files. But that was an issue in my theme.

Awesome work.

File: packages/block-library/src/table/style.scss
Comment (jasmussen) [reply packages/block-library/src/table/style.scss]: Fixed-width layout has to work everywhere because it's a user-set option.

But the table doesn't have to be responsive out of the box, no, so okay to remove the width, min-width, and overflow rules and put those in theme.scss 👍 👍

**Review by noisysocks — COMMENTED** [review]
Looking good! Will test this in a bit 🙂

File: packages/block-library/src/table/index.js
Comment (talldan) [reply packages/block-library/src/table/index.js]: 👍 

Given that, I'm guessing that deprecated blocks don't care about newly added attributes? Would they just ignore them? There aren't any in this PR, but I do have some in another PR.

File: packages/block-library/src/table/index.js
Comment (jasmussen) [reply packages/block-library/src/table/index.js]: I feel like by putting these styles in `theme.scss` we are already making themes opt in to this, and I also feel like we _need_ a responsive play for any block that comes with Gutenberg. Even by making this opt in we're already watering down that aspect a bit. 

If that requires a `div` to be compatible, I think that's okay, but as is always the case I'm okay with being overruled by @mtias or @karmatosed

**Comment by noisysocks** [comment]
Everything works great in my testing except for this step:

> 3. Observed that the user can scroll horizontally in the preview to view the entire table.

screen shot 2018-09-25 at 14 04 00

I'm using the default WordPress theme.

**Review by noisysocks — APPROVED** [review]
Works great in my testing!

**Comment by talldan** [comment]
Thanks for the review! Pushed commit f7a8fc3 to make those recommended changes. Will wait on an answer to the thread above before merging.

**Pull request closed without merging**
